main/src/ca/uwaterloo/flix/language/phase/Resolver.scala

```diff
@@ -1961,49 +1955,46 @@ object Resolver {
   /**
     * Performs name resolution on the given pattern `pat0` in the namespace `ns0`.
     */
-  private def resolvePattern(pat0: NamedAst.Pattern, env: ListMap[String, Resolution], ns0: Name.NName, root: NamedAst.Root): Validation[ResolvedAst.Pattern, ResolutionError] = {
+  private def resolvePattern(pat0: NamedAst.Pattern, env: ListMap[String, Resolution], ns0: Name.NName, root: NamedAst.Root)(implicit sctx: SharedContext): ResolvedAst.Pattern = {
 
-    def visit(p0: NamedAst.Pattern): Validation[ResolvedAst.Pattern, ResolutionError] = p0 match {
+    def visit(p0: NamedAst.Pattern): ResolvedAst.Pattern = p0 match {
       case NamedAst.Pattern.Wild(loc) =>
-        Validation.success(ResolvedAst.Pattern.Wild(loc))
+        ResolvedAst.Pattern.Wild(loc)
 
       case NamedAst.Pattern.Var(sym, loc) =>
-        Validation.success(ResolvedAst.Pattern.Var(sym, loc))
+        ResolvedAst.Pattern.Var(sym, loc)
 
       case NamedAst.Pattern.Cst(cst, loc) =>
-        Validation.success(ResolvedAst.Pattern.Cst(cst, loc))
+        ResolvedAst.Pattern.Cst(cst, loc)
 
       case NamedAst.Pattern.Tag(qname, pat, loc) =>
         lookupTag(qname, env, ns0, root) match {
-          case Result.Ok(c) => mapN(visit(pat)) {
-            case p => ResolvedAst.Pattern.Tag(Ast.CaseSymUse(c.sym, qname.loc), p, loc)
-          }
-          case Result.Err(e) => Validation.toSoftFailure(ResolvedAst.Pattern.Error(loc), e)
+          case Result.Ok(c) =>
+            val p = visit(pat)
+            ResolvedAst.Pattern.Tag(Ast.CaseSymUse(c.sym, qname.loc), p, loc)
+          case Result.Err(error) =>
+            sctx.errors.add(error)
```

Comment:
This is not duplicating errors right?

Reply:
I don't think so... it would have been duplicated before then too

Reply:
Also, `lookupTag` returns a restult without adding anything to `sctx` so it's up to the caller what to do with the error and that's why it's added here

Reply:
OK: let us go for it.